### PR TITLE
Fix disconnections 

### PR DESCRIPTION
**Pull request by mberechetbidfx**
The API was disconnecting from BidFX services upon submitting a large number of subscriptions. This was caused because the price updates received were larger than a single socket read. Fixed by repeating socket reads until the entire price map is received.

### CODE REVIEW

File: tests/pixie/utils_test/test_varint.py [hunk 1]
```diff
@@ -138,8 +138,14 @@ def test_decode_encode_varint_number2(self):
...
-            int.to_bytes(i, byteorder="little", length=1)
-            for i in encode_varint(int_value)
+            int.to_bytes(i, byteorder="little", length=1) for i in encode_varint(int_value)
         ]
         mock_socket.recv = lambda x: value.pop(x - 1)
         return mock_socket
+
+    def test_reads_from_socket_until_requested_length_fulfilled(self):
+        mock_socket = mock.Mock()
+        values = [b"ult", b"res"]
```
Comment (psweeny) [inline tests/pixie/utils_test/test_varint.py:148]: Would be more readable to not reverse the items and use .pop(-1) or call .reverse() at list initialisation.

File: tests/pixie/utils_test/test_varint.py [hunk 2]
```diff
@@ -138,8 +138,14 @@ def test_decode_encode_varint_number2(self):
...
         mock_socket = mock.Mock()
         value = [
-            int.to_bytes(i, byteorder="little", length=1)
-            for i in encode_varint(int_value)
+            int.to_bytes(i, byteorder="little", length=1) for i in encode_varint(int_value)
         ]
         mock_socket.recv = lambda x: value.pop(x - 1)
         return mock_socket
+
+    def test_reads_from_socket_until_requested_length_fulfilled(self):
```
Comment (psweeny) [inline tests/pixie/utils_test/test_varint.py:146]: Add a test with a packet larger than expected: 'res', 'ul', 'tsarina'. This should still return 'results'.

**Review by psweeny — CHANGES_REQUESTED** [review]

File: bidfx/pricing/_pixie/pixie_provider.py [hunk 3]
```diff
@@ -107,7 +108,7 @@ def _session_connection_attempt(self):
             self._publish_provider_status(ProviderStatus.READY)
             self._price_server_read_loop()
         except Exception as e:
-            log.warning(f"connection attempt failed due to: {e}")
+            log.warning(f"connection attempt failed due to: {traceback.format_exc()}")
```
Comment (psweeny) [inline bidfx/pricing/_pixie/pixie_provider.py:111]: I'd like to understand what this traceback call provides. I hope it is not a stack trace. That would be inappropriate for a message intended for client consumption.

File: bidfx/pricing/_pixie/util/varint.py [hunk 4]
```diff
@@ -96,7 +96,12 @@ def decode_varint_from_socket(socket_connection):
 
 
 def read_bytes(socket_connection, length):
-    byte_ = socket_connection.recv(length)
-    if byte_ == b"":
-        raise socket.error("end of socket stream")
-    return byte_
+    result = b""
+
+    while len(result) != length:
```
Comment (psweeny) [inline bidfx/pricing/_pixie/util/varint.py:101]: Although not wrong, I don't like the use of != as there could be a potential error left uncaught should you run past the length. Better to explicitly use < as an extra guard against an infinite loop.

**Review by psweeny — CHANGES_REQUESTED** [review]

File: bidfx/pricing/_pixie/pixie_provider.py (hunk 3 above)
Comment (mberechetbidfx) [reply bidfx/pricing/_pixie/pixie_provider.py:111]: it was for a stack trace, I thought the initial error message was too brief -- will reverse it

**Review by psweeny — APPROVED** [review]
All good.
Thanks

**Pull request merged**